syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

package common;

service Api {
  rpc ListApis (ListApisInput) returns (ListApisOutput) {}
  rpc GetSystemInfo(google.protobuf.Empty) returns (SystemInfo) {}
}

message ListApisInput {
  optional string search = 1;
}

message ListApisOutput {
  repeated string apis = 1;
}

message Space {
  uint64 total = 1;
  uint64 used = 2;
  uint64 free = 3;
}

message SystemInfo {
  google.protobuf.Timestamp timestamp = 1;
  uint64 uptime = 2;
  Space ram = 3;
  Space swap = 4;
}